plugins/zenoh-plugin-rest/src/lib.rs

```diff
@@ -46,47 +47,71 @@ lazy_static::lazy_static! {
 }
 const RAW_KEY: &str = "_raw";
 
-fn payload_to_json(payload: Payload) -> String {
-    payload
-        .deserialize::<String>()
-        .unwrap_or_else(|_| format!(r#""{}""#, b64_std_engine.encode(payload.contiguous())))
+#[derive(Serialize, Deserialize)]
+struct JSONSample {
+    key: String,
+    value: serde_json::Value,
+    encoding: String,
+    time: Option<String>,
 }
 
-fn sample_to_json(sample: Sample) -> String {
-    format!(
-        r#"{{ "key": "{}", "value": {}, "encoding": "{}", "time": "{}" }}"#,
-        sample.key_expr.as_str(),
-        payload_to_json(sample.payload),
-        sample.encoding,
-        if let Some(ts) = sample.timestamp {
-            ts.to_string()
-        } else {
-            "None".to_string()
+pub fn base64_encode(data: &[u8]) -> String {
+    use base64::engine::general_purpose;
+    general_purpose::STANDARD.encode(data)
+}
+
+fn payload_to_string(payload: &Payload) -> String {
+    String::from_utf8(payload.contiguous().to_vec()).unwrap_or(base64_encode(&payload.contiguous()))
```

Comment:
Why not using the `.deserialize()` as initially done?

Reply:
Or even https://github.com/eclipse-zenoh/zenoh/blob/d808ba29321c00e60329806ec5e7eafde25acfdf/zenoh/src/payload.rs#L561?

Reply:
Didn't knew about it

Reply:
Updated code to use `StringOrBase64::from` and then dereferecing to `String`.

A `clone()` becomes necessary but should be fine.